alpha/template/basic/basic.go

```diff
@@ -68,14 +61,57 @@ func (t Template) Render(ctx context.Context, reader io.Reader) (*declcfg.Declar
 	return cfg, nil
 }
 
+// Schema implements the template.Template interface
+func (t *basicTemplate) Schema() string {
+	return schema
+}
+
+// Factory implements the template.TemplateFactory interface
+type Factory struct{}
+
+// CreateTemplate implements the template.TemplateFactory interface
+func (f *Factory) CreateTemplate(renderBundle template.BundleRenderer) template.Template {
```

Comment:
given that we call `New()` from Factory, should we actually have `New()` function public?